lib/wadler/print.rb

```diff
@@ -115,7 +121,16 @@ def nest(indent, open_obj = '', close_obj = '',
     #
     # @return [Nil]
     def text(value, width: value.length)
-      tokens << Oppen.string(value, width:)
+      if config.trim_trailing_whitespaces? && value.match(/((?:#{Regexp.escape(whitespace)})+)\Z/)
```

Comment:
Isn't this regex overkill?  Don't you mean `end_with?`?  Sure, in the body you'd have to loop over `remove_suffix`.  But that saves the need for regexes.

Reply:
Let's take Firas's input on the matter.